src/Tasks.UnitTests/TaskFactoryUtilities_Tests.cs

```diff
@@ -0,0 +1,115 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using Microsoft.Build.Shared;
+using Microsoft.Build.Shared.FileSystem;
+using Microsoft.Build.Utilities;
+using Shouldly;
+using Xunit;
+
+#nullable disable
+
+namespace Microsoft.Build.UnitTests
+{
+    public sealed class TaskFactoryUtilities_Tests
+    {
+        [Fact]
+        public void CreateProcessSpecificTaskDirectory_ShouldCreateValidDirectory()
+        {
+            // Act
+            string directory = TaskFactoryUtilities.CreateProcessSpecificTaskDirectory();
+
+            // Assert
+            directory.ShouldNotBeNull();
+            Directory.Exists(directory).ShouldBeTrue();
+            directory.ShouldContain(TaskFactoryUtilities.InlineTaskTempDllSubPath);
+            directory.ShouldContain($"pid_{EnvironmentUtilities.CurrentProcessId}");
+        }
+
+        [Fact]
+        public void GetTemporaryTaskAssemblyPath_ShouldReturnValidPath()
+        {
+            // Act
+            string assemblyPath = TaskFactoryUtilities.GetTemporaryTaskAssemblyPath();
+
+            // Assert
+            assemblyPath.ShouldNotBeNull();
+            assemblyPath.ShouldEndWith(".dll");
+            Path.GetDirectoryName(assemblyPath).ShouldContain(TaskFactoryUtilities.InlineTaskTempDllSubPath);
+        }
+
+        [Fact]
+        public void CreateLoadManifest_ShouldCreateFileWithDirectories()
+        {
+            using (var env = TestEnvironment.Create())
+            {
+                // Arrange
+                var tempAssemblyFile = env.CreateFile(".dll");
+                var directories = new List<string> { "dir1", "dir2", "dir3" };
+
+                // Act
+                string manifestPath = TaskFactoryUtilities.CreateLoadManifest(tempAssemblyFile.Path, directories);
+
+                // Assert
+                manifestPath.ShouldBe(tempAssemblyFile.Path + TaskFactoryUtilities.InlineTaskLoadManifestSuffix);
+                File.Exists(manifestPath).ShouldBeTrue();
+
+                string[] manifestContent = File.ReadAllLines(manifestPath);
+                manifestContent.Length.ShouldBe(3);
+                manifestContent.ShouldContain("dir1");
+                manifestContent.ShouldContain("dir2");
+                manifestContent.ShouldContain("dir3");
+            }
+        }
+
+        [Fact]
+        public void CreateLoadManifest_WithNullAssemblyPath_ShouldThrow()
```

Comment:
That's a very defensive test. There are similar tests in our repo, but they rather are for MSBuild syntax and not-so-trivial cases, fixing the expected behavior. I wonder whether it is an overkill to have those here for the internal utilities class. Let's ask team.